starfish/test/full_pipelines/api/test_merfish.py

```diff
@@ -179,53 +180,47 @@ def test_merfish_pipeline_cropped_data():
         low_passed.numpy_array[5, 0, 0, 40:50, 45:55]
     )
 
-    # scale the data by the scale factors
-    scale_factors = {
-        (t[Indices.ROUND], t[Indices.CH]): t['scale_factor']
-        for t in experiment.extras['scale_factors']
-    }
-    for indices in low_passed._iter_indices():
-        data = low_passed.get_slice(indices)[0]
-        scaled = data / scale_factors[indices[Indices.ROUND.value], indices[Indices.CH.value]]
-        low_passed.set_slice(indices, scaled)
+    sc_filt = ScaleByPercentile(p=90)
```

Comment:
I think we need to adjust this filter a little bit to have this make sense. As it stands, the filter is _scaling by the max_value_ which is directly counter to what it says it wants to do. The below diff would fix that by enabling above-max-value intensities to get clipped. In this regime, all values are actually scaled by the stated percentile. 

```diff
$ g diff starfish/image/_filter/scale_by_percentile.py
diff --git a/starfish/image/_filter/scale_by_percentile.py b/starfish/image/_filter/scale_by_percentile.py
index 07e9d72..de57a7e 100644
--- a/starfish/image/_filter/scale_by_percentile.py
+++ b/starfish/image/_filter/scale_by_percentile.py
@@ -5,6 +5,7 @@ import numpy as np
 
 from starfish.stack import ImageStack
 from ._base import FilterAlgorithmBase
+from .util import preserve_float_range
 
 
 class ScaleByPercentile(FilterAlgorithmBase):
@@ -30,16 +31,21 @@ class ScaleByPercentile(FilterAlgorithmBase):
             "--p", default=100, type=int, help="scale images by this percentile")
 
     @staticmethod
-    def scale(image: np.ndarray, p: int) -> np.ndarray:
+    def scale(image: np.ndarray, p: int, clip=True) -> np.ndarray:
         """Clip values of img below and above percentiles p_min and p_max
 
         Parameters
         ----------
         image : np.ndarray
             image to be scaled
-
         p : int
             each image in the stack is scaled by this percentile. must be in [0, 100]
+        clip : bool
+            if True, clip any values that end up out of scale as a result of filtering. This has
+            the effect of eliminating outlier intensities. If False, the top-end value will be
+            used to rescale the dynamic range to [0, 1], which may mitigate or eliminate the effect
+            of scaling in the case that the image was already saturated (contains 1-valued pixels).
+            (Default = True)
 
         Notes
         -----
@@ -54,11 +60,12 @@ class ScaleByPercentile(FilterAlgorithmBase):
         """
         v = np.percentile(image, p)
 
-        # asking for a float percentile clipping value from an integer image will
-        # convert to float, so store the dtype so it can be restored
-        dtype = image.dtype
         image = image / v
-        return image.astype(dtype)
+        if clip:
+            image[image > 1] = 1
+        image = preserve_float_range(image)
+
+        return image
 
     def run(
             self, stack: ImageStack, in_place: bool=True, verbose: bool=False,
```